### PR TITLE
Add support for `prefer album artist` option to ListenBrainz

### PR DESCRIPTION
**Pull request by staticssleever668**
Reference #274.

CC #273.

### CODE REVIEW

File: src/scrobbler/listenbrainzscrobbler.cpp [hunk 1]
```diff
@@ -429,7 +433,7 @@ void ListenBrainzScrobbler::UpdateNowPlaying(const Song &song) {
   title = title.remove(Song::kTitleRemoveMisc);
 
   QJsonObject object_track_metadata;
-  if (song.albumartist().isEmpty() || song.albumartist().compare(Song::kVariousArtists, Qt::CaseInsensitive) == 0) {
```
Comment (jonaski) [inline src/scrobbler/listenbrainzscrobbler.cpp:432]: albumartist can be empty, so artist needs to be used for artist_name in that case.

File: src/scrobbler/listenbrainzscrobbler.cpp [hunk 2]
```diff
@@ -552,7 +556,7 @@ void ListenBrainzScrobbler::Submit() {
     QJsonObject object_listen;
     object_listen.insert("listened_at", QJsonValue::fromVariant(item->timestamp_));
     QJsonObject object_track_metadata;
-    if (item->albumartist_.isEmpty() || item->albumartist_.compare(Song::kVariousArtists, Qt::CaseInsensitive) == 0) {
+    if (!prefer_albumartist_ || item->albumartist_.compare(Song::kVariousArtists, Qt::CaseInsensitive) == 0) {
```
Comment (jonaski) [inline src/scrobbler/listenbrainzscrobbler.cpp:559]: Same here

**Review by jonaski — CHANGES_REQUESTED** [review]

File: src/scrobbler/listenbrainzscrobbler.cpp (hunk 1 above)
Comment (staticssleever668) [reply src/scrobbler/listenbrainzscrobbler.cpp:432]: Right, I'm not even sure why I removed this check...

File: src/scrobbler/listenbrainzscrobbler.h [hunk 3]
```diff
@@ -120,6 +120,8 @@ class ListenBrainzScrobbler : public ScrobblerService {
   QTimer timer_submit_;
   bool submit_error_;
 
+  bool prefer_albumartist_;
```
[the remark this answers is not in the document]
Comment (staticssleever668) [reply src/scrobbler/listenbrainzscrobbler.h:123]: Done.

**Review by jonaski — APPROVED** [review]

**Pull request merged**